Libraries/Opc.Ua.Client/Session.cs

```diff
@@ -6059,18 +6052,19 @@ private CallMethodRequestCollection CreateCallRequestsForResendData(IEnumerable<
         /// <summary>
         /// Create the publish requests for the active subscriptions.
         /// </summary>
-        private void RestartPublishing()
+        public void StartPublish(int timeout)
         {
-            int publishCount = 0;
-            lock (SyncRoot)
+            int publishCount = GetMinPublishRequestCount(true);
+
+            if (m_tooManyPublishRequests > 0 && publishCount > m_tooManyPublishRequests)
             {
-                publishCount = GetMinPublishRequestCount(true);
+                publishCount = m_tooManyPublishRequests;
             }
```

Comment:
m_tooManyPublishRequests was not used (yet) due to unknown behavior in case of temporary server overload, it could lead to a very low number. Lower than what is configured for MinPublishRequestCount.